### PR TITLE
ci: twisterlib: Limit the regex used to extract overlays

### PR DESCRIPTION
**Pull request by piwe-nordic**
This commit changes regular expression used to merge overlays specified for test samples in twister tests.

Overlays with any prefixes will be ignored in merging mechanism and passed as is to cmake command.

Signed-off-by: Piotr Węgliński <piotr.weglinski@nordicsemi.no>

### CODE REVIEW

**Review by hakehuang — DISMISSED** [review]

**Review by PerMac — DISMISSED** [review]

**Review by PerMac — DISMISSED** [review]

**Comment by nashif** [comment]
> OVERLAY_CONFIG with prefix 

what does that mean, can you give an example?

**Review by hakehuang — DISMISSED** [review]

**Comment by piwe-nordic** [comment]
[nashif](https://github.com/nashif)
Is it possible to merge this change? Thank You

**Comment by nashif** [comment]
> i.e. we use -Dremote_OVERLAY_CONFIG="path/to/file.conf". It is intended for child image and it is parsed in further steps on cmake. Current implementation of twister changs cmake call and it is impossible to use this configuration in a way we intend.

it is not clear from the commit message what you are trying to do, so if I understand correctly, you do not want twister to handle anything with a prefix and foce parsing to only apply on OVERLAY_CONFIG?

**Comment by piwe-nordic** [comment]
@nashif Yes. That's precisely my goal.

**Comment by rakons** [comment]
@nashif I had hit the issue with one of our tests. It basically seems like oversight anyway to have a rule that glues together anything that finishes with OVERLAY_CONFIG and puts it into OVERLAY_CONFIG variable. It should be exactly OVERLAY_CONFIG, not [anything]OVERLAY_CONFIG.

**Review by rakons — DISMISSED** [review]

**Comment by nashif** [comment]
> @nashif Yes. That's precisely my goal.

then please refine your commit message, it means exactly the opposite to me:

> In cases overlays where passed with prefix it was ignored
and overlays where merged regardless.
In cases of using overlays for i.e. child images
OVERLAY_CONFIG with prefix was not able to be used
as it would be applied to base image regards of any prefixes.

**Review by nashif — CHANGES_REQUESTED** [review]
please refine commit message

**Comment by piwe-nordic** [comment]
Changed the commit message.

File: scripts/pylib/twister/twisterlib/runner.py [hunk 1]
```diff
@@ -784,7 +784,7 @@ def cmake(self):
 
         # merge overlay files into one variable
```
Comment (PerMac) [inline scripts/pylib/twister/twisterlib/runner.py:785]: I believe it is worth to explain the modified mechanism in a comment. E.g. expand the comment with: "Overlays with prefixes won't be merged but pass to cmake as they are.

Comment (piwe-nordic) [reply scripts/pylib/twister/twisterlib/runner.py:785]: Will this commit message be ok:  
This commit changes regular expression used to merge overlays 
specified for test samples in twister tests.

Overlays without any prefixes will be merged and passed to cmake.
Overlays with prefixes won't be merged but passed to cmake as they are.

Comment (rakons) [reply scripts/pylib/twister/twisterlib/runner.py:785]: What I wish to point out - that in the form before changes, all overlays with prefixes was merged but to the one variable without prefix. So the mechanism was broken by design, and this commit fixes it, but I believe we cannot use word "fix" in a commit.

Comment (piwe-nordic) [reply scripts/pylib/twister/twisterlib/runner.py:785]: Yes - it was broken. But per my understanding child images are not use in the upstream so it was never an issue here. Could You please accept the changes if it is OK now.

**Review by PerMac — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by rakons** [comment]
@nashif Can we merge this change? Do you think it might provide any issues with sysbuild?

**Review by rakons — APPROVED** [review]

**Comment by piwe-nordic** [comment]
@nashif You are still marked as `requesting change` and merge is hence blocked. How should i address this?

**Review by nashif — APPROVED** [review]

**Pull request merged**